resources/views/formfields/relationship.blade.php

```diff
@@ -155,7 +159,9 @@ class="form-control @if(isset($options->taggable) && $options->taggable == 'on')
                 >
 
                         @php
-                            $selected_values = isset($dataTypeContent) ? $dataTypeContent->belongsToMany($options->model, $options->pivot_table)->pluck($options->table.'.'.$options->key)->all() : array();
+                            $selected_values = isset($dataTypeContent) ? $dataTypeContent->belongsToMany($options->model, $options->pivot_table)->get()->map(function ($item, $key) use ($options) {
+                                return $item->{$options->label};
```

Comment:
This needs to be `return $item->{$options->key}` because it is used in the check below, in line 173:
`@if(in_array($relationshipOption->{$options->key}, $selected_values)){{ 'selected="selected"' }}@endif`

Reply:
The way it's currently implemented, belongsToMany select2 fields always stay empty, because it tries to look for the keys in `$selected_values`, but `$selected_values` stores the labels.

Reply:
Good spot. It will take care of #3457, too. I'll push another commit.